core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -374,4 +385,107 @@ static TableMetadata commit(TableOperations ops, UpdateTableRequest request) {
+    request.validate();
+
+    ViewMetadata viewMetadata = request.metadata();
+    ViewBuilder viewBuilder =
+        catalog
+            .buildView(TableIdentifier.of(namespace, request.name()))
+            .withSchema(viewMetadata.schema())
+            .withProperties(viewMetadata.properties())
+            .withDefaultNamespace(viewMetadata.currentVersion().defaultNamespace())
+            .withDefaultCatalog(viewMetadata.currentVersion().defaultCatalog());
+    viewMetadata.currentVersion().representations().stream()
+        .filter(r -> r instanceof SQLViewRepresentation)
+        .map(r -> (SQLViewRepresentation) r)
+        .forEach(r -> viewBuilder.withQuery(r.dialect(), r.sql()));
+    View view = viewBuilder.create();
+
+    return viewResponse(view);
+  }
+
+  private static LoadViewResponse viewResponse(View view) {
+    ViewMetadata metadata = baseView(view).operations().current();
+    return ImmutableLoadViewResponse.builder()
+        .metadata(metadata)
+        .metadataLocation(metadata.metadataFileLocation())
+        .build();
+  }
+
+  public static LoadViewResponse loadView(ViewCatalog catalog, TableIdentifier viewIdentifier) {
+    View view = catalog.loadView(viewIdentifier);
+    return viewResponse(view);
+  }
+
+  public static LoadViewResponse updateView(
+      ViewCatalog catalog, TableIdentifier ident, UpdateTableRequest request) {
+    View view = catalog.loadView(ident);
+    ViewMetadata metadata = commit(baseView(view).operations(), request);
+
+    return ImmutableLoadViewResponse.builder()
+        .metadata(metadata)
+        .metadataLocation(metadata.metadataFileLocation())
+        .build();
+  }
+
+  public static void renameView(ViewCatalog catalog, RenameTableRequest request) {
+    catalog.renameView(request.source(), request.destination());
+  }
+
+  public static void dropView(ViewCatalog catalog, TableIdentifier viewIdentifier) {
+    boolean dropped = catalog.dropView(viewIdentifier);
+    if (!dropped) {
+      throw new NoSuchViewException("View does not exist: %s", viewIdentifier);
+    }
+  }
+
+  static ViewMetadata commit(ViewOperations ops, UpdateTableRequest request) {
+    AtomicBoolean isRetry = new AtomicBoolean(false);
+    try {
+      Tasks.foreach(ops)
+          .retry(COMMIT_NUM_RETRIES_DEFAULT)
+          .exponentialBackoff(
+              COMMIT_MIN_RETRY_WAIT_MS_DEFAULT,
+              COMMIT_MAX_RETRY_WAIT_MS_DEFAULT,
+              COMMIT_TOTAL_RETRY_TIME_MS_DEFAULT,
+              2.0 /* exponential */)
+          .onlyRetryOn(CommitFailedException.class)
+          .run(
+              taskOps -> {
+                ViewMetadata base = isRetry.get() ? taskOps.refresh() : taskOps.current();
+                isRetry.set(true);
+
+                // apply changes
+                ViewMetadata.Builder metadataBuilder = ViewMetadata.buildFrom(base);
+                request.updates().forEach(update -> update.applyTo(metadataBuilder));
```

Comment:
Why does this skip validating requirements?

Reply:
based on https://github.com/apache/iceberg/pull/8147#discussion_r1274225077 we decided that we don't need view requirements (for now), hence this code wasn't validating any requirements.

Reply:
I see we're currently running through the validations on line 486 but if I'm not mistaken right now they're all just no-ops, is that right @nastra ? 

Few requirements that come to mind:

1.) For creating a view, there probably needs to be a requirement similar to table's `AssertTableDoesNotExist` but for views, `AssertViewDoesNotExist`. Similar principle for ReplaceView and verifying the UUID as expected.


2.) I don't see a metadata update for AddViewRepresentation but I'd imagine that should validate that the SQL dialect does not already exist for the specified version. I forgot where we landed on that discussion, We do want to prevent that right?

3.) A requirement for the schema ID to make sure that is unchanged from the base metadata?

Reply:
I think we will need to support the AssertUUID requirement right? It also seems good to run them even if we expect there to be none. Otherwise, when we add a requirement we'll have to go back and find out why it doesn't actually work. And similarly, if someone sends a requirement that is not allowed for views (but can be parsed for tables) we need to attempt to run it so that we will throw the exception that it isn't supported.

Reply:
I've opened https://github.com/apache/iceberg/pull/8831 to introduce `AssertUUID` (and deprecate `AssertTableUUID`) so that we can use it both for tables and views. Alternatively, we could also introduce `AssertViewUUID` with the same functionality as `AssertTableUUID`

Reply:
I'd like to get this working with `AssertTableUUID`, independent of adding `AssertUUID`. It doesn't bother me to use the table classes temporarily.

I don't think we need `AssertCurrentSchemaID`.